src/dbnode/storage/index_query_concurrent_test.go

```diff
@@ -254,7 +253,14 @@ func testNamespaceIndexHighConcurrentQueries(
 						r index.QueryResults,
 						logFields []opentracinglog.Field,
 					) (bool, error) {
-						timeoutWg.Wait()
+						// block until the query will be canceled due to timeout.
```

Comment:
@robskillington this test broke because it was relying on the original flow control, that has now changed. now there is a no longer a goroutine waiting for the timeout and returning. instead there is a goroutine watching for the timeout and interrupting the workers (which ultimately results in a return). however, that doesn't work with the previous flow since `block.Query` never runs to get interrupted.

this patch makes the test go green, but unclear if it has the same coverage as before. I'm not sure what the previous test was actually trying to accomplish. It appears it was trying to stress test racing to close the ctx. However, not even sure if that's possible anymore since we are properly interrupting the workers _before_ returning from the rpc call.